matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/SessionModule.kt

```diff
@@ -290,17 +290,17 @@ internal abstract class SessionModule {
             return matrixConfiguration.cryptoConfig
         }
 
-        @JvmStatic
         @Provides
+        @JvmStatic
         @SessionScope
-        fun providesAuthApi(
+        fun providesRefreshTokenAPI(
                 @Unauthenticated okHttpClient: Lazy<OkHttpClient>,
                 retrofitFactory: RetrofitFactory,
                 homeServerConnectionConfig: HomeServerConnectionConfig
-        ): AuthAPI {
+        ): RefreshTokenAPI {
```

Comment:
Can you have `retrofit: Retrofit` as the only parameter?

Reply:
The `okHttpClient` here is unauthenticated. Would I have to provide another unauthenticated version of `Retrofit`? is there any example of where we do that?

Reply:
I see your (good) point about potential endless loop

Reply:
Yeah you can keep it like that it's ok